src/compiler/clang.rs

```diff
@@ -60,29 +62,31 @@ impl CCompilerImpl for Clang {
                      executable: &Path,
                      parsed_args: &ParsedArguments,
                      cwd: &Path,
-                     env_vars: &[(OsString, OsString)])
+                     env_vars: &[(OsString, OsString)],
+                     may_dist: bool)
                      -> SFuture<process::Output> where T: CommandCreatorSync
     {
-        gcc::preprocess(creator, executable, parsed_args, cwd, env_vars)
+        gcc::preprocess(creator, executable, parsed_args, cwd, env_vars, may_dist)
     }
 
-    fn compile<T>(&self,
-                  creator: &T,
-                  executable: &Path,
-                  parsed_args: &ParsedArguments,
-                  cwd: &Path,
-                  env_vars: &[(OsString, OsString)])
-                  -> SFuture<(Cacheable, process::Output)>
-        where T: CommandCreatorSync
+    fn generate_compile_commands(&self,
+                                path_transformer: &mut dist::PathTransformer,
+                                executable: &Path,
+                                parsed_args: &ParsedArguments,
+                                cwd: &Path,
+                                env_vars: &[(OsString, OsString)])
+                                -> Result<(CompileCommand, Option<dist::CompileCommand>, Cacheable)>
     {
-        gcc::compile(creator, executable, parsed_args, cwd, env_vars)
+        gcc::generate_compile_commands(path_transformer, executable, parsed_args, cwd, env_vars)
     }
 }
 
-static ARGS: [(ArgInfo, gcc::GCCArgAttribute); 8] = [
+pub static ARGS: [(ArgInfo, gcc::GCCArgAttribute); 8] = [
     take_arg!("--serialize-diagnostics", String, Separated, PassThrough),
     take_arg!("--target", String, Separated, PassThrough),
-    take_arg!("-Xclang", String, Separated, PassThrough),
+    // TODO: should be extracted and reprocessed, though bear in mind some
+    // flags are not valid under a -Xclang
+    take_arg!("-Xclang", String, Separated, TooHard),
```

Comment:
It felt a bit dubious to me to just pass these through blindly.

Reply:
I didn't realize this was even here, honestly. I don't think I've ever seen this used outside of clang-cl.